### PR TITLE
[TINKERPOP-2949] added PBiPredicate interface

### PR DESCRIPTION
**Pull request by vkagamlyk**
Introduced `PBiPredicate` interface to reduce serialization issues like [INKERPOP-2945](https://issues.apache.org/jira/browse/TINKERPOP-2945)

JIRA https://issues.apache.org/jira/browse/TINKERPOP-2949

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/PBiPredicate.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.tinkerpop.gremlin.process.traversal;
+
+import java.util.function.BiPredicate;
+
+/**
+ * Marker interface for predefined {@code BiPredicate} predicates that can be used in {@code Predicate}}.
```
Comment (spmallette) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/PBiPredicate.java:25]: could you update the javadoc to say a few words about why this interface was needed, what it helps to do, what problems it helps avoid, etc.?

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -26,6 +26,7 @@ image::https://raw.githubusercontent.com/apache/tinkerpop/master/docs/static/ima
 This release also includes changes from <<release-3-6-XXX, 3.6.XXX>>.
 
 * Removed `connectOnStartup` configuration option from gremlin-javascript.
+* Added marker interface `PBiPredicate` for predefined predicates.
```
Comment (spmallette) [inline CHANGELOG.asciidoc:29]: should there be provider Upgrade Documentation for this? there are providers with custom predicates. knowing this change was available and when to use it, might be useful to them?

**Comment by Cole-Greer** [comment]
LGTM VOTE +1

**Comment by xiazcy** [comment]
VOTE +1

**Pull request merged**